framework/features/PulseFeature.h

```diff
@@ -0,0 +1,66 @@
+#ifndef PULSEFEATURE_H
+#define PULSEFEATURE_H
+
+#include "Feature.h"
+
+template<const char* const name, uint16_t gpio, bool invert = false, uint16_t damper = 0, uint16_t max_duration = 0>
+class PulseFeature : public Feature<name> {
+
+protected:
+    using Feature<name>::LOG;
+
+public:
+    PulseFeature(Device* device) :
+            Feature<name>(device),
+            lastChange(RTC.getRtcSeconds()) {
+        pinMode(gpio, OUTPUT);
+        digitalWrite(gpio, invert);
+        this->switchBackTimer.initializeMs(100, TimerDelegate(&PulseFeature::switchOff, this));
+        this->registerSubscription("pulse", Device::MessageCallback(&PulseFeature::onMessageReceived, this));
+
+        LOG.log("Initialized");
+    }
+
+    void doPulse(const int ms) {
+        switchBackTimer.setIntervalMs(ms);
+        digitalWrite(gpio, !invert);
+        switchBackTimer.start(false);
+    }
+
+protected:
+    virtual void publishCurrentState() {};
+
+private:
+    void onMessageReceived(const String& topic, const String& message) {
+        const uint32_t now = RTC.getRtcSeconds();
+
+        if (damper > 0) {
+            if (this->lastChange + damper > now) {
+                this->publish("error", "cooldown active, try again later", false);
+                LOG.log("message ignored because still in damping");
+                return;
+            }
+        }
+        const int value = atoi(message.c_str());
```

Comment:
By the way: I couldn't find a nice Int -> String-Function, too. I'd like to report the actual parsed timeout instead of the raw message received on the feedback-topic, but with snprintf it did not compile. Any idea?